components/Banner/examples/supplemental/MDCBannerView.m

```diff
@@ -54,124 +69,312 @@ - (instancetype)initWithCoder:(NSCoder *)aDecoder {
+  trailingButton.translatesAutoresizingMaskIntoConstraints = NO;
+  trailingButton.backgroundColor = UIColor.whiteColor;
+  [buttonContainerView addSubview:trailingButton];
+  _trailingButton = trailingButton;
 }
 
 #pragma mark - Property Setter and Getter
 
-- (void)setText:(NSString *)text {
-  self.textLabel.text = text;
+- (BOOL)isImageSet {
+  return self.imageView.image != nil && !self.imageView.hidden;
 }
 
-- (NSString *)text {
-  return self.textLabel.text;
-}
+#pragma mark - KVO
 
-- (void)setImage:(UIImage *)image {
-  if (image) {
-    _image = image;
-    if (!self.iconImageViewContainer) {
-      CGRect iconImageContainerFrame =
-          CGRectMake(0, 0, kIconImageContainerSideLength, kIconImageContainerSideLength);
-      UIView *iconImageViewContainer = [[UIView alloc] initWithFrame:iconImageContainerFrame];
-      UIImageView *iconImageView = [[UIImageView alloc] initWithImage:image];
-      CGRect iconImageFrame = CGRectMake(0, 0, kIconImageSideLength, kIconImageSideLength);
-      iconImageView.frame = iconImageFrame;
-      iconImageView.contentMode = UIViewContentModeScaleAspectFit;
-      iconImageView.clipsToBounds = YES;
-      [iconImageViewContainer addSubview:iconImageView];
-      self.iconImageViewContainer = iconImageViewContainer;
-      self.iconImageView = iconImageView;
-      [self addSubview:iconImageViewContainer];
-    } else {
-      self.iconImageView.image = image;
-    }
-  } else {
-    _image = nil;
-    [self.iconImageViewContainer removeFromSuperview];
-    self.iconImageViewContainer = nil;
-  }
-  [self setNeedsLayout];
-}
+// TODO: KVO trailingButton.hidden and call `setNeedsUpdateConstraints:`.
 
-- (void)setNumberOfButtons:(NSUInteger)numberOfButtons {
-  NSAssert(numberOfButtons <= kNumberOfButtonsLimit,
-           @"%@ class doesn't support more than %lu buttons", NSStringFromClass([self class]),
-           (unsigned long)kNumberOfButtonsLimit);
-  if (numberOfButtons != self.buttons.count) {
-    NSMutableArray *mutableButtons = [self.buttons mutableCopy];
-    if (numberOfButtons > self.buttons.count) {
-      for (NSUInteger index = self.buttons.count; index < numberOfButtons; ++index) {
-        MDCButton *button = [[MDCButton alloc] init];
-        [mutableButtons addObject:button];
-        [self addSubview:button];
-      }
-    } else if (numberOfButtons < self.buttons.count) {
-      for (NSUInteger index = numberOfButtons; index < self.buttons.count; ++index) {
-        MDCButton *lastObject = [mutableButtons lastObject];
-        [lastObject removeFromSuperview];
-        [mutableButtons removeLastObject];
+#pragma mark - UIView overrides
+
+- (CGSize)sizeThatFits:(CGSize)size {
+  MDCBannerViewLayoutStyle style = [self styleForSizeToFit:size];
+  CGFloat frameHeight = 0.0f;
+  switch (style) {
+    case MDCBannerViewLayoutSingleLineStyle: {
+      frameHeight += kTopPaddingSmall + kBottomPadding;
+      [self.textLabel sizeToFit];
```

Comment:
it is a odd to call sizeToFit from within a sizeThatFits. More often sizeThatFits calls more sizeThatFits of its subviews.

Think of it this way, sizeThatFits is asking the banner view what size would you be if you were told you had this much space. We do not expect the subviews (label and buttons) to change size when we ask this.

Reply:
Thanks for suggestion. I was using `sizeToFit` and then getting the frame, but it is right that changing the actual frame is unnecessary. 

Addressed in 070341d.